src/ui/public/vis/editors/default/default.js

```diff
@@ -127,6 +137,32 @@ const defaultEditor = function ($rootScope, $compile) {
             catch (e) {} // eslint-disable-line no-empty
           }, true);
 
+          $scope.$watch(() => {
+            return $scope.vis.getCurrentState(false);
+          }, (newState) => {
+            // figure out what changed and copy into editor state
+            const recursiveCopyChanged = (src, org, dst) => {
+              for (const prop in src) {
+                if (typeof src[prop] === 'object') {
+                  if (!dst[prop]) {
+                    dst[prop] = _.cloneDeep(src[prop]);
+                    org[prop] = _.cloneDeep(src[prop]);
+                  } else {
+                    recursiveCopyChanged(src[prop], org[prop], dst[prop]);
+                  }
+                } else {
+                  if (src[prop] !== org[prop]) {
+                    org[prop] = src[prop];
+                    dst[prop] = src[prop];
+                    lockDirty = true;
```

Comment:
Why do we only set `lockDirty` here but not on line 150?

Reply:
bug, thanks for finding this